packages/core/src/checkout-buttons/strategies/googlepay/googlepay-button-strategy.ts

```diff
@@ -160,13 +185,55 @@ export default class GooglePayButtonStrategy implements CheckoutButtonStrategy {
         throw new InvalidArgumentError();
     }
 
+    private async _createBuyNowCart({ buyNowInitializeOptions }: BuyNowInitializeOptions) {
+        if (
+            typeof buyNowInitializeOptions?.getBuyNowCartRequestBody === 'function' &&
+            this._cartRequestSender
+        ) {
+            const cartRequestBody = buyNowInitializeOptions.getBuyNowCartRequestBody();
+
+            if (!cartRequestBody) {
+                throw new MissingDataError(MissingDataErrorType.MissingCart);
+            }
+
+            try {
+                const { body: cart } = await this._cartRequestSender.createBuyNowCart(
+                    cartRequestBody,
+                );
+
+                return cart;
+            } catch (error) {
+                throw new BuyNowCartCreationError();
+            }
+        }
+    }
+
     @bind
-    private async _handleWalletButtonClick(event: Event): Promise<void> {
+    private async _handleWalletButtonClick(
+        event: Event,
+        { buyNowInitializeOptions }: BuyNowInitializeOptions,
+        currencyCode?: string,
+    ): Promise<void> {
         event.preventDefault();
 
-        const cart = this._store.getState().cart.getCartOrThrow();
+        this._buyNowCart = await this._createBuyNowCart({ buyNowInitializeOptions });
+
+        const cart = this._buyNowCart || this._store.getState().cart.getCartOrThrow();
         const hasPhysicalItems = getShippableItemsCount(cart) > 0;
 
+        if (this._buyNowCart && currencyCode) {
+            const payloadToUpdate: { currencyCode: string; totalPrice: string } = {
```

Comment:
Is there any reason to keep this interface here? (`{ currencyCode: string; totalPrice: string }`)
As far as I see, we have a simple object here and Typescript can define its data structure by itself

Reply:
yeap
TypeScript can define it is data structure by itself
 i have fixed it
thanks